### DISCUSSION TITLE
(Linux) Installing Dreambooth fills up my root partition

### DISCUSSION THREAD

**Discussion by izz-j** [d8ahazard/sd_dreambooth_extension#948] [Q&A] [answered]
I know that dreambooth installs in ~/stable-diffusion-ui/extensions/
But when I click the install button in WebUI, it also eats up all of my storage space in my / partition. 
I am guessing it must be dependencies. But I can't see what sub-directory these dependencies are being installed. I would like them to be installed into my ~/home user partition instead of / 
And when I delete the dreambooth directory, ofcourse these dependencies are still installed somewhere in my system.

Anyone know where these dependencies are installed? I am running Debian Buster.

**Comment by Arilziem** [comment, marked as answer]
The only dependencies being installed should be those by pip (which normally install into `./venv`) and some additional models that are downloaded into `~/.cache`. So what you are describing should not happen and is probably related to something with your environment.

Thoughts:
* Are you manually using pip without activating your venv? And maybe even as superuser? That would probably put stuff in `/root`, which is not in `/home` unless you do ... things).
* Did you set any custom path options in `webui[-user].bat`? A path you set might be wrong (although other things should not be writeable outside `/home` as a regular user, with a few notable exceptions)
* Are you _sure_ sure that your home is actually on a different partition? I know weird question but unless you are doing that pip thing or the path thing, that is the only other thing that comes to mind.

**Reply by izz-j** [reply]
Thank you for your response. I solved it by checking the requirements text file and installing each of them manually by including '--user' flag. It seems that by default they were being installed into my root partition instead of my home parition (~/.local)

**Reply by Arilziem** [reply]
You should not need to do that at all. Normally, dependencies are installed automatically when you launch the webui. On first run, it creates the venv and downloads all necessary dependencies. If you need to specify `--user` with pip, you likely haven't activated your venv (`source venv/bin/activate`), which will then install everything into the venv subdirectory (but again, you usually don't need that). When using --user (or system install), you might also run into issues later on, when the requirements are changed and older packages become incompatible, but then can't be changed by the webui because they are not in `venv`.

As a sidenote, you should _never_ do this with superuser/root rights. You don't need it for pip, and you definitely can't trust the webui and all its extensions to run as root on your system.

I would strongly recommend you remove the packages from your user install, run the normal webui-user.sh (which will install everything into `venv`), then run the webui-dreambooth.sh which will install the remaining things automatically. Good luck :)

**Reply by izz-j** [reply]
Thank you. I removed all the packages and followed your instruction. I previously did not use venv. Now that I installed everything using venv I have a much cleaner install.
